### PR TITLE
adding rpm builds for ppc64le and s390x

### PR DESCRIPTION
**Pull request by utzb**
Allows to build rpms on ppc64le and s390x for users of openSUSE and CentOS/ClefOS environments. Used via build-rpm target of make.sh.

Signed-off-by: Christy Perez <christy@linux.vnet.ibm.com>
Signed-off-by: Christopher Jones <tophj@linux.vnet.ibm.com>
Signed-off-by: Dominik Dingel <dingel@de.ibm.com>
Signed-off-by: Thorsten Winkler <thorsten.winkler@de.ibm.com>
Signed-off-by: Utz Bacher <utz.bacher@de.ibm.com>

**- What I did**
Add Dockerfiles and according build/generate shell scripts

**- How to verify it**
hack/make.sh build-rpm on ppc64le/s390x

**- Description for the changelog**
Enable build of rpms for ppc64le and s390x

### CODE REVIEW

**Comment by tophj-ibm** [comment]
I've tested the ppc64le builds, and the s390x packages lgtm

**Review by tophj-ibm — APPROVED** [review]

**Comment by utzb** [comment]
s390x tested as well

**Comment by vieux** [comment]
ping @andrewhsu

**Comment by andrewhsu** [comment]
This PR is definitely helpful for paving a way towards eventually supporting more ppc64le and s390x architecture packages.

I'd say bring this in knowing it is more of a contrib type submission.

**Comment by thaJeztah** [comment]
ping @tianon PTAL as well ❤️

**Review by thaJeztah — APPROVED** [review]
LGTM

File: contrib/builder/rpm/s390x/clefos-base-s390x-7/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#
+# THIS FILE IS AUTOGENERATED; SEE "contrib/builder/rpm/s390x/generate.sh"!
+#
+
+FROM sinenomine/clefos-base-s390x
```
Comment (tianon) [inline contrib/builder/rpm/s390x/clefos-base-s390x-7/Dockerfile]: This isn't exactly official -- is ClefOS something we want to support? (appears to be a Z-specific CentOS/RHEL derivative?)

File: contrib/builder/rpm/s390x/generate.sh [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+fi
+versions=( "${versions[@]%/}" )
+
+for version in "${versions[@]}"; do
+	echo "${versions[@]}"
+	distro="${version%-*}"
+	suite="${version##*-}"
+	case "$distro" in
+		*opensuse*)
+            from="opensuse/s390x:tumbleweed"
```
Comment (tianon) [inline contrib/builder/rpm/s390x/generate.sh]: Whitespace throughout this file looks pretty inconsistent -- should be all-tabs.

File: contrib/builder/rpm/s390x/clefos-base-s390x-7/Dockerfile (hunk 1 above)
Comment (utzb) [inline contrib/builder/rpm/s390x/clefos-base-s390x-7/Dockerfile]: That is true; these guys are not part of the CentOS organization (which is with Red Hat now), so they are not allowed to name it that way. Technically, they follow CentOS 1:1. CentOS themselves does not cover s390x today.

File: contrib/builder/rpm/s390x/generate.sh (hunk 2 above)
Comment (utzb) [inline contrib/builder/rpm/s390x/generate.sh]: rrrrr... Yes, let me fix that. Thanks.

**Comment by utzb** [comment]
latest change fixes up blank vs. white spaces in contrib/builder/rpm/s390x/generate.sh as @tianon suggested (thanks!)

**Comment by thaJeztah** [comment]
What shall we do with the `clefos` version? Is this a popular choice on s390x?

**Comment by utzb** [comment]
@thaJeztah it is actually the s390x equivalent to CentOS, and technically is it, just for Sine Nomine (the folks behind it) not part of CentOS organization, different naming. So technically, a good choice. Non-technically, Sine Nomine is a close partner of IBM, and has a long record of very active participation in the Linux on s390x community. Similar to x86, CentOS/ClefOS gets meaning through its proximity to Red Hat.
Looking at the user base on the mainframe, almost all clients bet on commercial distributions for production (the major players being SLES, RHEL, and Ubuntu).

As the submitted of the PR, I may be a bit biased ;-) , but IMHO having ClefOS brings s390x closer to what is typically done and available on x86. It can also be a good base for further commercialization, i.e. be even useful when Docker builds EE deliverables. If the usage landscape changes, we can always discuss adjusting this choice, but I believe s390x has accelerated to embrace more Open Source projects over the past 5 years (ClefOS being one sign of it), so I hope this momentum continues.

**Comment by utzb** [comment]
Hello @tianon @thaJeztah , how would you like to proceed on this PR? I don't have a good feeling of when these images are moved over to being part of official (in which case I'll be happy to submit another PR, adapting to that)

**Comment by tianon** [comment]
Given that @andrewhsu thinks they're OK, I'm fine with these going forward. :+1: :heart:

**Comment by thaJeztah** [comment]
I take that as a LGTM

**Pull request merged**